src/sql/workbench/api/browser/mainThreadDataProtocol.ts

```diff
@@ -672,4 +678,8 @@ export class MainThreadDataProtocol extends Disposable implements MainThreadData
 
 		return undefined;
 	}
+
+	public $mark(name: string): void {
```

Comment:
Perf should be available to the ext host as well - e.g. https://github.com/Microsoft/azuredatastudio/blob/8a3d08f0dec1cdf47d7fc442f90fb2227b828375/src/vs/workbench/api/common/extHostExtensionService.ts#L211 - so you shouldn't need to do this over the RPC channel

Reply:
I initially tried to call `perf.mark()` directly in extHost process and pass needed marks back to renderer process (as marks are stored in each process) via RPC channel in one batch but that required too many irrelevant-ish changes in query runner code so I changed to the current implementation.

I also did a quick test with the delay caused by this cross-process RPC mark call and the delay is around 0-2ms and I think this is good enough for query perf tracing. 
(Time stamps on the left side were the real timestamp logged in the extHost process and the right ones were the ones we got via RPC $mark)

![image](https://user-images.githubusercontent.com/21186993/222338675-36ce4312-240d-4bc2-83d4-bdb3037aa844.png)

Reply:
So for my understanding - the problem is that when you try to collect the perf marks in the renderer process to print to the logs it doesn't include the ext host ones?

That makes sense I guess, although unfortunate. 

I would say then that at the very least you should make a separate RPC proxy for perf and move the mark call to that - that way at least other things can use it instead of having it contained to just dataprotocol.

Reply:
That is correct, marks from extHost are not included in the result of `getMarks()` if we call it in renderer.

I've moved the changes to a separate RPC proxy.

Apart from `mark()` itself, I will also try to add another function to send marks over in one batch which I think is better but that will be another PR later.